ddtrace/internal/datadog/profiling/ddup.pyx

```diff
@@ -60,31 +60,30 @@ IF UNAME_SYSNAME == "Linux" and UNAME_MACHINE == "x86_64":
         void ddup_upload()
 
     def init(
-            service: str,
+            service: Optional[str],
             env: Optional[str],
             version: Optional[str],
             tags: Optional[typing.Dict[str, str]],
             max_nframes: Optional[int],
             url: Optional[str]) -> None:
 
         # Try to provide a ddtrace-specific default service if one is not given
-        if not service:
-            service = DEFAULT_SERVICE_NAME
+        service = service if service else DEFAULT_SERVICE_NAME
```

Comment:
```suggestion
        service = service or DEFAULT_SERVICE_NAME
```

Reply:
Good catch, `x if x` doesn't add any explicitness/precision over the short-circuiting or.  Taken in bd2698b2143261a51d170b55420861dfe5ee3f36